crates/bvs-vault-cw20-tokenized/src/contract.rs

```diff
@@ -0,0 +1,705 @@
+            CombinedExecuteMsg::TransferFrom {
+                owner,
+                recipient,
+                amount,
+            } => execute_transfer_from(deps, env, info, owner, recipient, amount),
+            CombinedExecuteMsg::SendFrom {
+                owner,
+                contract,
+                amount,
+                msg,
+            } => execute_send_from(deps, env, info, owner, contract, amount, msg),
+            _ => {
+                // Extended execute msg set are exhausted in entry point already
+                // Base cw20 execute msg are also exhausted in other match arm
+                // So this means someone is trying to call a non-supported message
+                Err(cw20_base::ContractError::Std(StdError::generic_err(
+                    "This message is not supported",
+                )))
+            }
+        }
+    }
+}
+
+/// Additional vault logic are built on top of the base CW20 contract via an extended execute msg set.
+/// The extended execute msg set is practically `bvs-vault-base` crate's execute msg set.
+mod vault_execute {
+    use crate::error::ContractError;
+    use bvs_vault_base::error::VaultError;
+    use bvs_vault_base::msg::{Recipient, RecipientAmount};
+    use bvs_vault_base::{
+        offset, router,
+        shares::{self, QueuedWithdrawalInfo},
+    };
+    use bvs_vault_cw20::token as UnderlyingToken;
+    use cosmwasm_std::{DepsMut, Env, Event, MessageInfo, Response, Timestamp};
+    use cw20_base::contract::execute_burn as receipt_token_burn;
+
+    /// This executes a transfer of assets from the `info.sender` to the vault contract.
+    ///
+    /// New receipt token are minted, based on the exchange rate, to `msg.recipient`.  
+    /// The `TOTAL_SHARE` (aka `TOTAL_SUPPLY`) in the vault is increased.
+    ///
+    /// ### CW20 Variant Warning
+    ///
+    /// Underlying assets that are not strictly CW20 compliant may cause unexpected behavior in token balances.
+    /// For example, any token with a fee-on-transfer mechanism is not supported.
+    ///
+    /// Therefore, we do not support non-standard CW20 tokens.
+    /// Vault deployed with such tokens will be blacklisted in the vault-router.
+    pub fn deposit_for(
+        mut deps: DepsMut,
+        env: Env,
+        info: MessageInfo,
+        msg: RecipientAmount,
+    ) -> Result<Response, ContractError> {
+        router::assert_whitelisted(&deps.as_ref(), &env)?;
+
+        let assets = msg.amount;
+
+        let (vault, new_receipt_tokens) = {
+            let underlying_token_balance = UnderlyingToken::query_balance(&deps.as_ref(), &env)?;
+            let receipt_token_supply =
+                cw20_base::contract::query_token_info(deps.as_ref())?.total_supply;
+            let vault = offset::VirtualOffset::new(receipt_token_supply, underlying_token_balance)?;
+
+            let new_receipt_tokens = vault.assets_to_shares(assets)?;
+
+            (vault, new_receipt_tokens)
+        };
```

Comment:
```suggestion
        let new_receipt_tokens = {
            let underlying_token_balance = UnderlyingToken::query_balance(&deps.as_ref(), &env)?;
            let receipt_token_supply =
                cw20_base::contract::query_token_info(deps.as_ref())?.total_supply;
            let vault = offset::VirtualOffset::new(receipt_token_supply, underlying_token_balance)?;

            let new_receipt_tokens = vault.assets_to_shares(assets)?;

            new_receipt_tokens
        };
```

Don't return vault to prevent confusion.